infra/src/main/scala/ch/epfl/scala/services/storage/sql/SqlRepo.scala

```diff
@@ -1,37 +1,56 @@
 package ch.epfl.scala.services.storage.sql
 
-import cats.effect.{IO, Resource}
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+import cats.effect.IO
 import ch.epfl.scala.index.model.Project
-import ch.epfl.scala.index.model.misc.GithubInfo
-import doobie.implicits._
-import ch.epfl.scala.index.newModel.{NewDependency, NewProject, NewRelease}
+import ch.epfl.scala.index.newModel.NewDependency
+import ch.epfl.scala.index.newModel.NewProject
+import ch.epfl.scala.index.newModel.NewRelease
 import ch.epfl.scala.services.DatabaseApi
-import ch.epfl.scala.services.storage.sql.tables.{
-  DependenciesTable,
-  GithubInfoTable,
-  ProjectTable,
-  ReleaseTable
-}
+import ch.epfl.scala.services.storage.sql.tables.DependenciesTable
+import ch.epfl.scala.services.storage.sql.tables.GithubInfoTable
+import ch.epfl.scala.services.storage.sql.tables.ProjectTable
+import ch.epfl.scala.services.storage.sql.tables.ProjectUserFormTable
+import ch.epfl.scala.services.storage.sql.tables.ReleaseTable
 import ch.epfl.scala.utils.DoobieUtils
-
-import scala.concurrent.Future
-import scala.concurrent.ExecutionContext.Implicits.global
+import doobie.implicits._
 
 class SqlRepo(conf: DbConf) extends DatabaseApi {
   private[sql] val (xa, flyway) = DoobieUtils.create(conf)
   def createTables(): IO[Unit] = IO(flyway.migrate())
   def dropTables(): IO[Unit] = IO(flyway.clean())
 
-  override def insertProject(project: NewProject): Future[NewProject] = {
+  override def insertProject(project: NewProject): Future[Unit] = {
     for {
-      p <- run(ProjectTable.insert, project)
+      _ <- run(ProjectTable.insert, project)
+      _ <- run(ProjectUserFormTable.insert(project), project.formData)
       _ <- project.githubInfo
         .map(run(GithubInfoTable.insert(project), _))
         .getOrElse(Future.successful())
-    } yield p
+    } yield ()
   }
 
-  def insertProject(project: Project): Future[NewProject] =
+  override def findProject(
+      projectRef: Project.Reference
+  ): Future[Option[NewProject]] =
+    for {
+      project <- run(ProjectTable.selectOne(projectRef.org, projectRef.repo))
+      userForm <- run(
+        ProjectUserFormTable.selectOne(projectRef.org, projectRef.repo)
+      )
+      githubInfoTable <- project
+        .map(p => run(GithubInfoTable.selectOne(p.organization, p.repository)))
+        .getOrElse(Future.successful(None))
```

Comment:
If this is too slow we can make it faster by running the 3 futures in parallel. (later)